src/devices/Hcsr04/Hcsr04.cs

```diff
@@ -90,39 +92,39 @@ private bool TryGetDistance(out double result)
             // 100ms was chosen because max measurement time for this sensor is around 24ms for 400cm
             // additionally we need to account 60ms max delay.
             // Rounding this up to a 100 in case of a context switch.
-            long hangTicks = Environment.TickCount + 100;
+            long hangTicks = DateTime.Now.Ticks + 100 * TimeSpan.TicksPerMillisecond;
             _timer.Reset();
 
             // Measurements should be 60ms apart, in order to prevent trigger signal mixing with echo signal
             // ref https://components101.com/sites/default/files/component_datasheet/HCSR04%20Datasheet.pdf
-            while (Environment.TickCount - _lastMeasurment < 60)
+            while (DateTime.Now.Ticks - _lastMeasurment < 60 * TimeSpan.TicksPerMillisecond)
             {
-                Thread.Sleep(TimeSpan.FromMilliseconds(Environment.TickCount - _lastMeasurment));
+                Thread.Sleep(TimeSpan.FromTicks(DateTime.Now.Ticks - _lastMeasurment));
             }
 
+            _lastMeasurment = DateTime.Now.Ticks;
+
             // Trigger input for 10uS to start ranging
             _controller.Write(_trigger, PinValue.High);
-            Thread.Sleep(TimeSpan.FromMilliseconds(0.01));
+            DelayHelper.DelayMicroseconds(10, true);
             _controller.Write(_trigger, PinValue.Low);
 
             // Wait until the echo pin is HIGH (that marks the beginning of the pulse length we want to measure)
             while (_controller.Read(_echo) == PinValue.Low)
             {
-                if (Environment.TickCount - hangTicks > 0)
+                if (DateTime.Now.Ticks - hangTicks > 0)
```

Comment:
perhaps we should add GetTicks (with inlining) to DelayHelpers in case we need to do this more often in the future and also have a way to convert ticks to seconds. The original code is fine here but seems it's not supported on nano. Also note that DateTime.Now will likely be broken if someone changes system time while app is running

Reply:
If someone will change the time, it's fine, it will just break 1 measure. Not a big deal :-) None of the sensors are perfect. They all can return wrong values time to time. This sensor has multiple retry because of the quality of the measurement anyway.
And same, I already changed the element but didn't save before pushing.